dataprofiler/profilers/histogram_utils.py

```diff
@@ -14,6 +14,218 @@
 )
 
 
+def rework_ptp(maximum, minimum):
+    """Peak-to-peak value of x.
+
+    This implementation avoids the problem of signed integer arrays having a
+    peak-to-peak value that cannot be represented with the array's data type.
+    This function returns an unsigned value for signed integer arrays.
+    """
+    return np.subtract(maximum, minimum)
+
+
+def rework_hist_bin_doane(profile):
+    """
+    Doane's histogram bin estimator.
+
+    Improved version of Sturges' formula which works better for
+    non-normal data. See
+    stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+
+    Parameters
+    ----------
+    x : array_like
+        Input data that is to be histogrammed, trimmed to range. May not
+        be empty.
+
+    Returns
+    -------
+    h : An estimate of the optimal bin width for the given data.
+    """
+    dataset_size = sum(profile._stored_histogram["histogram"]["bin_counts"])
```

Comment:
We can instead use the `profile.sampling_size`.

Reply:
AttributeError: 'IntColumn' object has no attribute 'sampling_size'